compiler/rustc_error_codes/src/error_codes/E0744.md

```diff
@@ -12,7 +12,11 @@ const _: i32 = {
 };
 ```
 
-At the moment, `for` is forbidden inside a `const`, `static`, or `const fn`.
+At the moment, `for` loops, and the `Try` operator (`?`) is forbidden inside a `const`, `static`, or `const fn`. Similarly, `match` expressions which desugarto `for` loops, code using the `Try` operator (`?`), or code using `await` are
```

Comment:
This seems odd - when would a `match` desugar to a loop? For loops desugar to 
```rust
let mut iter = iter.into_iter();
loop {
  match iter.next() {
    Some(elem) => loop_body(elem),
    None => break,
  }
}
```

Reply:
See [compiler/rustc_passes/src/check_const.rs](https://github.com/rust-lang/rust/blob/master/compiler/rustc_passes/src/check_const.rs) and [rustc_hir::MatchSource](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.MatchSource.html), I might have interpreted it wrong.

Reply:
I think it's the other way around :sweat_smile: